feat: Added supported waveshare displays

Vishveshwara opened the pull request:
## Summary by Sourcery

Introduce comprehensive support for various Waveshare NFC ePaper models by adding new display classes, updating asset references to WebP, and centralizing display logic into a unified module

New Features:
- Add support for multiple Waveshare NFC ePaper displays (2.13", 2.9b, 2.7", 4.2", 7.5", 7.5" HD)
- Implement color modes and dithering pipelines for the Waveshare 2.9" B display

Enhancements:
- Consolidate individual Waveshare display modules into a single waveshare_displays file
- Replace PNG assets with WebP for all Waveshare display images

hpdang commented:
This looks good to me. I wonder if we should rename the first two E-paper that we had to something more recognizable by the users perhaps the branch name of those? 

![WhatsApp Image 2025-08-20 at 4 34 22 PM](https://github.com/user-attachments/assets/ae7de181-8233-4554-98f5-90baabf2c94b)

Vishveshwara commented:
> @Vishveshwara do we still need one or it is already included in your other PR #159

Yes this is a seperate PR just for adding the displays and it needs to be merged , PR #159 only changes the inner working for the  transfer logic

hpdang commented:
@Vishveshwara please resolve conflicts

Vishveshwara commented:
![Screenshot_20250828-132924.png](https://github.com/user-attachments/assets/cbf83984-169c-4f4e-b6dc-fc89ef5ec576)

Vishveshwara commented:
@hpdang @Dhruv1797 I have removed the display chip , Added the SKU number for each display as it is present in the box of the waveshare hardware, changed the text from driver to SDK for waveshare displays.

Vishveshwara commented:
The SKU code on the waveshare box

![IMG_20250828_134324.jpg](https://github.com/user-attachments/assets/bab012f9-0a61-442d-bddd-176e84b9e0a4)

Vishveshwara commented:
@hpdang This can be merged
